### PR TITLE
[IMP] display message subject

### PR DESCRIPTION
**Pull request by oomsveta**

### CODE REVIEW

File: addons/mail/static/src/new/thread/message.js [hunk 1]
```diff
@@ -80,6 +80,22 @@ export class Message extends Component {
...
+        const prefixList = ["re", "fw", "fwd"];
+        // match any prefix as many times as possible
+        const prefixesRegexp = new RegExp(`((${prefixList.join("|")}):\\s*)+`);
+        const cleanedSubject = this.message.subject.toLowerCase().replace(prefixesRegexp, "");
+        const cleanedOriginThreadName = this.originThread.name.toLowerCase().trim();
+        return cleanedSubject === cleanedOriginThreadName;
+    }
+
+    get originThread() {
+        return this.messaging.threads[this.message.resId];
```
Comment (seb-odoo) [inline addons/mail/static/src/new/thread/message.js]: this is not correct, to find a thread we need both resModel and resId

File: addons/mail/static/src/new/thread/message.js [hunk 2]
```diff
@@ -80,6 +80,22 @@ export class Message extends Component {
...
+    get isSubjectSimilarToOriginThreadName() {
+        if (!this.message.subject || !this.originThread || !this.originThread.name) {
+            return false;
+        }
+        const prefixList = ["re", "fw", "fwd"];
+        // match any prefix as many times as possible
+        const prefixesRegexp = new RegExp(`^((${prefixList.join("|")}):\\s*)+`);
+        const cleanedSubject = this.message.subject.toLowerCase().replace(prefixesRegexp, "");
+        const cleanedOriginThreadName = this.originThread.name.toLowerCase().trim();
+        return cleanedSubject === cleanedOriginThreadName;
```
Comment (seb-odoo) [inline addons/mail/static/src/new/thread/message.js]: If you think the code is better this way it would be nice to make this change in master so that diff in this PR is reduced to see the code moved.

\+ adding tests to cover edge cases to make sure the change is keeping the behavior

File: addons/mail/static/src/new/thread/message.xml [hunk 3]
```diff
@@ -66,6 +66,7 @@
                                 'o-mail-message-body align-self-start text-break mb-0 rounded-end-3 rounded-bottom-3 p-3': !message.isNote,
                                 'flex-grow-1': state.isEditing,
                             }">
+                                <em t-if="message.subject and !message.isSubjectSimilarToOriginThreadName" class="o-mail-message-subject mb-1 me-2">Subject: <t t-esc="message.subject"/></em>
```
Comment (seb-odoo) [inline addons/mail/static/src/new/thread/message.xml:69]: could be self-closing em tag

**Review by seb-odoo — APPROVED** [review]

**Comment by seb-odoo** [comment]
conflict to resolve but otherwise this is ok
